### DISCUSSION TITLE
OE pin voltage - related to brightness?

### DISCUSSION THREAD

**Discussion by juniorradu** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#291] [Q&A] [answered]
Hello Mrfaptastic, 
I am trying to understand how does the brightness work on this matrix led panels.
I observed that the voltage on the OE pin is 5V if the panel is powered with 5V and the HUB75 cable is not connected, but once you connect the cable and turn on the ESP32, the OE pin voltage is reduced to 3.3V (lowest brightness) and then 2V, 1V, 0.3V.. as you increase the brightness by software. 
1. Is it not dangerous for the ESP to connect 5V to the GPIO? (maximum 3.3V accepted). I just burnt an ESP32 and am trying to find out the cause.
2. How does the ESP32 indicate the Panel to reduce the brightness? 

Thanks!
![photo_2022-06-23 12 21 45](https://user-images.githubusercontent.com/79705541/175278871-2f31d4f8-0b73-4e00-99f1-1762d27297d9.jpeg)
![photo_2022-06-23 12 21 57](https://user-images.githubusercontent.com/79705541/175278874-c4c42a7b-96c5-4654-8daf-c4f040b65088.jpeg)
![photo_2022-06-23 12 22 01](https://user-images.githubusercontent.com/79705541/175278877-e12b2adf-1e1b-4d92-a418-c9d1a249dc7d.jpeg)

**Comment by board707** [comment, marked as answer]
Main purpose of OE pin is not for brightness control. It used for switching panels lights ON and OFF.
OE means "Output enable", it has inverse logic, so OE = HIGH means panell OFF and OE = LOW is panel ON.
but if one swithing OE quickly (for example, by PWM) - it can be used for control of brightness.

**Reply by mrcodetastic** [reply]
As what @board707 said.

But also, the digitial multimeter is getting confused by the rapid on/off PWM signal on the OE pin and interpreting as if it's an analogue value between 0-3.3volts. It is not - It's either only 0 or 3.3volts, just dont really quickly to give the human perception of 'brightness'.
